homeassistant/components/recorder/__init__.py

```diff
@@ -230,8 +230,8 @@ def purge_ticker(event):
                 self.queue.task_done()
                 continue
 
-            entity_id = event.data.get(ATTR_ENTITY_ID)
-            domain = event.data.get(ATTR_DOMAIN)
+            entity_id = str(event.data.get(ATTR_ENTITY_ID))
```

Comment:
👍  Events that are not STATE_CHANGE dont have entity_ids.

New commit should handle this better